tests/arch/x86/hybrid/src/smpboot.c

```diff
@@ -0,0 +1,154 @@
+/* Copyright (c) 2023 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cpuid.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/kernel_structs.h>
+
+BUILD_ASSERT(CONFIG_SMP);
+BUILD_ASSERT(CONFIG_X86_64);
+
+#define CPUID_MASK_WORD  0xFF
+#define CPUID_MASK_NIBLE 0x0F
+#define STACKSZ          2048
+char stack[STACKSZ];
+
+volatile bool mp_flag[CONFIG_MP_MAX_NUM_CPUS];
+
+struct k_thread cpu_thr;
+K_THREAD_STACK_DEFINE(thr_stack, STACKSZ);
+
+extern void z_smp_start_cpu(int id);
+
+static int cpu_info_get(struct x86_cpu_info *info)
+{
```

Comment:
I wonder could we test the function you wrote in `cpu.c` :  ` x86_update_cpu_info()`